plugins/src/test/java/opengrok/auth/plugin/LdapServerTest.java

```diff
@@ -79,9 +81,24 @@ public void testSetGetUsername() {
 
     @Test
     public void testIsReachable() throws IOException, InterruptedException, URISyntaxException {
-        // Start simple TCP server on port 6336. It has to be > 1024 to avoid BindException
-        // due to permission denied.
-        int testPort = 6336;
+        final int testPortBase = 6336;    // It has to be > 1024 to avoid BindException due to permission denied.
+        final int randomRange = 49152;
+        int triesCount = 0;
+        final int MAX_PORT_TRIES = 20;
+        Random rand = new Random();
+        int testPort;
+        while (true) {
+            testPort = testPortBase + rand.nextInt(randomRange);
+            if (PortChecker.available(testPort)) {
+                break;
+            }
+            if (++triesCount > MAX_PORT_TRIES) {
+                throw new RuntimeException("Could not find an available port after " +
+                        MAX_PORT_TRIES + " tries");
+            }
+        }
+
+        // Start simple TCP server on test port.
         InetAddress localhostAddr = InetAddress.getLocalHost();
         ServerSocket serverSocket = new ServerSocket(testPort, 1, localhostAddr);
```

Comment:
Why not just specify `0` for port here?
From https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/ServerSocket.html#%3Cinit%3E(int,int,java.net.InetAddress)
> A port number of 0 means that the port number is automatically allocated, typically from an ephemeral port range. This port number can then be retrieved by calling getLocalPort.

Reply:
That's not a bad idea. Let me try that.

Reply:
Looks like it works. This is probably better solution because there is no time-of-use-time-of-check gap.